src/hooks/useFetchPlugins.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import { galleryApiFetch } from "../services/apiService";
+import { Plugin } from "../interfaces/Plugin";
+
+/**
+ * Fetches plugins from the backend api.
+ *
+ * @param url url to fetch plugins from
+ * @param pageSize number of plugins to fetch each page
+ * @param pageNum page number to fetch for
+ * @param searchQuery search query for filtering plugins
+ */
+function useFetchPlugins(
+  url: string,
+	pageSize: number,
+	pageNum: number,
+	searchQuery?: string
+  ) {
+    const [plugins, setPlugins] = useState<Plugin[]>([]);
+      const [isLoading, setIsLoading] = useState(false);
+      const [error, setError] = useState("");
+    useEffect(function(){
+      
+      const fetchPlugins = async() => {
```

Comment:
can you disintegrate the fetching function and the hook ? This may open new maintenance issues in the future. Separate the state controls and the fetching logic, then call the function. (check this pr:  https://github.com/React-ChatBotify/gallery-website/pull/78)